File: Capitulo3- Funcoes/Funcoes.Funcional/Funcoes/FuncoesAninhadas.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Funcoes
{
    class FuncoesAninhadas
    {
        public void EscreveSeNumeroEParOuImpar(int numero)
        {
            bool verificaSeONumeroEPar() => numero % 2 == 0;
            void escreveNumeroPar() => Console.WriteLine(string.Format("O número {0} é par", numero));
            void escreveNumeroImpar() => Console.WriteLine(string.Format("O número {0} é impar", numero));

            if (verificaSeONumeroEPar())
                escreveNumeroPar();
            else
                escreveNumeroImpar();
        }
    }
}

File: Capitulo3- Funcoes/Funcoes.Funcional/Funcoes.Funcional.NetFramework/FuncoesAninhadas.fs


module FuncoesAninhadas

let escreveSeNumeroEParOuImpar numero =
    let verificaSeONumeroEPar() = numero % 2 = 0
    let escreveNumeroPar() = printfn "O número %i é par" numero
    let escreveNumeroImpar() = printfn "O número %i é impar" numero

    if verificaSeONumeroEPar() then
        escreveNumeroPar()
    else
        escreveNumeroImpar()

